### PR TITLE
DQN copy_model_parameters memory leak

### PR DESCRIPTION
**Pull request by Kismuz**
Hello @dennybritz , thank you for maintaining this project.
Couple of things with `Deep Q Learning Solution`:
1. There was a caveat in the implementation of `copy_model_parameters()` function: update ops are redefined every call, leaving a lot of disconnected subgraphs, which lead to subtile but inevitable memory consumption build up. It will show up while iterating over ~500k since a lot of memory is already consumed by replay buffer. 
Proposed change will eliminate the problem.

2. Added process health monitor to tensorboard summaries since it’s a concern with mid/low-end machines, especially with high replay memory settings.

### CODE REVIEW

**Comment by dennybritz** [comment]
Thanks! I confirm I've seen the memory leak, but didn't know where it was.

**Comment by wetliu** [comment]
Hello. May I ask if there is any benchmark of this version of code compared with the original DQN paper, such as the final epsilon=0.05 performance? Thank you!

**Comment by Kismuz** [comment]
Original DQN results where obtained using some additional features like reward and gradient clipping, carefully chosen optimizer learning rate annealing and 'soft parameter copying'; so it's not quite correct to compare. 
This implementation is correct and can be viewed as 'basic' algorithm, making DQN principles easy to understand; it can be tweaked to achieve maximum possible performance.

**Comment by wetliu** [comment]
Thank you for letting me know! I appreciate your response. I will change/test it and see how good it can perform.
